src/gui.rs

```diff
@@ -331,32 +339,56 @@ impl MyApp {
                     }
 
                     let t_fmt = |x, _range: &RangeInclusive<f64>| format!("{:4.2} s", x);
-                    let signal_plot = Plot::new("data")
-                        .height(height)
-                        .width(width)
-                        .legend(Legend::default())
-                        .x_grid_spacer(log_grid_spacer(10))
-                        .y_grid_spacer(log_grid_spacer(10))
-                        .x_axis_formatter(t_fmt)
-                        .min_size(vec2(50.0, 100.0));
-
-                    let plot_inner = signal_plot.show(ui, |signal_plot_ui| {
-                        for (i, graph) in graphs.into_iter().enumerate() {
-                            // this check needs to be here for when we change devices (not very elegant)
-                            if i < self.serial_devices.labels[self.device_idx].len() {
-                                signal_plot_ui.line(
-                                    Line::new(PlotPoints::Owned(graph))
-                                        .name(&self.serial_devices.labels[self.device_idx][i]),
-                                );
+
+                    let plots_ui = ui.vertical(|ui| {
+                        for graph_idx in 0..self.number_of_plots {
+                            if graph_idx != 0 {
+                                ui.separator();
                             }
+
+                            let signal_plot = Plot::new(format!("data-{graph_idx}"))
+                                .height(plot_height)
+                                .width(width)
+                                .legend(Legend::default())
+                                .x_grid_spacer(log_grid_spacer(10))
+                                .y_grid_spacer(log_grid_spacer(10))
+                                .x_axis_formatter(t_fmt);
+
+                            let plot_inner = signal_plot.show(ui, |signal_plot_ui| {
+                                for (i, graph) in graphs.iter().enumerate() {
+                                    // this check needs to be here for when we change devices (not very elegant)
+                                    if i < self.serial_devices.labels[self.device_idx].len() {
+                                        signal_plot_ui.line(
+                                            Line::new(PlotPoints::Owned(graph.to_vec())).name(
+                                                &self.serial_devices.labels[self.device_idx][i],
+                                            ),
+                                        );
+                                    }
+                                }
+                            });
+
+                            self.plot_location = plot_inner.response.rect;
                         }
+                        let separator_response = ui.separator();
```

Comment:
change this to
```rust
let separator_response = ui.separator();
    let separator = ui
        .interact(
            separator_response.rect,
            separator_response.id,
            Sense::click_and_drag(),
        ).on_hover_cursor(egui::CursorIcon::ResizeVertical);
    
    let resize_y = separator
        .drag_delta()
        .y;
    
    if separator.double_clicked() {
        self.plot_serial_display_ratio = 0.45;
    }
    self.plot_serial_display_ratio = (self.plot_serial_display_ratio
        + resize_y / panel_height)
        .clamp(0.1, 0.9);
    
    ui.add_space(spacing);
```

so that we can double click the separator to reset it to the default